### PR TITLE
Exercise2

### PR DESCRIPTION
**Pull request by dlopezm**

### CODE REVIEW

File: exercises/00-init/src/modules/root/components/header.js [hunk 1]
```diff
@@ -1,14 +1,12 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-class Header extends React.Component {
-    render() {
-      return <h1>{this.props.title}</h1>;
-    }
+function Header (props) {
+      return <h1>{props.title}</h1>;
+}
```
Comment (martisekpetr) [inline exercises/00-init/src/modules/root/components/header.js]: It is customary to use arrow functions for function components, i.e. 
```const Header = props => <h1>{ props.title }</h1>;```
or optionally
```const Header = ({ title }) => <h1>{ title }</h1>;```
Not that using a classic function is wrong, but you will never see it in any Salsita project :)

File: exercises/00-init/src/modules/root/components/users-list.js [hunk 2]
```diff
@@ -1,34 +1,26 @@
...
-class UsersList extends React.Component {
-    constructor() {
-        super();
-        this.state = { 
-            users : []
-        }
-    }
-
-    render() {
+function UsersList (props) {
```
Comment (martisekpetr) [inline exercises/00-init/src/modules/root/components/users-list.js]: Same here. You can place the long return value in `()` parentheses, e.g.
``` 
const UsersList = props => (
  <div>
    ...
  </div>
);

File: exercises/00-init/src/modules/root/components/users-list.js [hunk 3]
```diff
@@ -1,34 +1,26 @@
...
-    addUser(firstName, lastName) {
-        this.setState( 
-            { users: [
-                ...this.state.users,
-                {id: this.state.users.length, firstName, lastName}
-            ]
-        })
-    }
+UsersList.propTypes = {
+    users: PropTypes.arrayOf(PropTypes.shape({
```
Comment (martisekpetr) [inline exercises/00-init/src/modules/root/components/users-list.js:18]: :100:

File: exercises/00-init/src/modules/root/components/root.js [hunk 4]
```diff
@@ -2,10 +2,10 @@ import React from 'react';
...
 
-const Root = () => (
+const Root = (props) => (
   <div>
-    <Header title="User Management" />
-    <UsersList/>
+    <Header title={props.title} />
+    <UsersList users={props.users} addUser={props.addUser}/>
   </div>
 );
```
Comment (martisekpetr) [inline exercises/00-init/src/modules/root/components/root.js:11]: Missing propTypes for Root component. No worries though, eslint would let you know if it was configured on this project :) Also, you basically never get any props on `<Root>` element, so :man_shrugging:

File: exercises/00-init/src/index.js [hunk 5]
```diff
@@ -5,7 +5,26 @@ import ReactDOM from 'react-dom';
...
+  title: "Amazing list of users",
+  users: []
+}
+
+function addUser(firstName, lastName) {
+  state.users = 
+  [
+    ...state.users,
+    { id: state.users.length, firstName, lastName }
+  ]
```
Comment (martisekpetr) [inline exercises/00-init/src/index.js:18]: :+1:

**Review by martisekpetr — APPROVED** [review]

File: exercises/00-init/src/modules/root/components/header.js (hunk 1 above)
Comment (dlopezm) [reply exercises/00-init/src/modules/root/components/header.js]: As before, I'm defaulting to the syntax I'm most familiar with. If I have to get familiar with a different one, I'll start now, by changing it as you suggest :)

File: exercises/00-init/src/modules/root/components/root.js (hunk 4 above)
Comment (dlopezm) [reply exercises/00-init/src/modules/root/components/root.js:11]: Nothing that can't be solved by a little copypasting ;).

**Pull request merged**